### PR TITLE
Adapt in-app messaging to native integration

### LINKED ISSUE #14267: Display different alert content if the admin turned on Sourcegraph on the code host level
Part of [RFC 221](https://docs.google.com/document/d/19f4xleYBU1zZZdqMmXlLmFxeR-fwEpOwTOgViOFOnyo/edit?usp=sharing): Improve discoverability/awareness of the browser extension

**Description**

Display [different alert content](https://www.figma.com/file/V5t7KOzC6PIX75L4dz8IFc/RFC-221-Browser-extension-improve-discoverability-and-awareness-Approved?node-id=558%3A17) if the admin turned on Sourcegraph on the code host level

**Design documentation**

- ‘Sourcegraph's code intelligence will follow you to your code host. Your site admin set up the Sourcegraph native integration for [code host]. Learn more or just try it out’
- possible for GitLab, Bitbucket Server, and Phabricator

### PR DESCRIPTION
**Pull request by felixfbecker**
Closes #14267. Implements the simple version of a setting.

### CODE REVIEW

File: doc/integration/bitbucket_server.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ For the Bitbucket Server plugin to then communicate with the Sourcegraph instanc
 }
 ```
 
+The site admin should also set `codeHost.nativeIntegrationInstalled` in [global settings](../admin/config/settings#editing-global-settings-for-site-admins) to ensure informational content for users in the Sourcegraph webapp references the native integration and not the browser extension.
```
Comment (tjkandala) [inline doc/integration/bitbucket_server.md:45]:
```suggestion
The site admin should also set `codeHost.nativeIntegrationInstalled` in [global settings](../admin/config/settings.md#editing-global-settings-for-site-admins) to ensure informational content for users in the Sourcegraph webapp references the native integration and not the browser extension.
```

File: doc/admin/external_service/gitlab.md [hunk 2]
```diff
@@ -74,6 +74,8 @@ The Sourcegraph instance's site admin must [update the `corsOrigin` site config
 }
 ```
 
+The site admin should also set `codeHost.nativeIntegrationInstalled` in [global settings](../config/settings#editing-global-settings-for-site-admins) to ensure informational content for users in the Sourcegraph webapp references the native integration and not the browser extension.
```
Comment (tjkandala) [inline doc/admin/external_service/gitlab.md:77]:
```suggestion
The site admin should also set `codeHost.nativeIntegrationInstalled` in [global settings](../config/settings.md#editing-global-settings-for-site-admins) to ensure informational content for users in the Sourcegraph webapp references the native integration and not the browser extension.
```

**Comment by marekweb** [comment]
I'm not sure if the intent of the flag is clear, from the description.

Is the intent that the site admin can set this flag to change what gets displayed to users, but it's independent of whether a native integration is actually installed?

In that case, would it be clearer if the setting name were framed as a display setting (e.g. along the lines of `codehost.showXYZ`) instead of `installed`?

I'm wondering if using the word `installed` in the setting sets an unclear expectation about whether this setting actually controls enabling/installing a native integration (and whether disabling this would disable an installed native integration)

**Comment by felixfbecker** [comment]
@marekweb I see your point, but I find it hard to come up with a name that encompasses all the places that are affected by this and the ways they are. Do you have a suggestion for a better name?

**Comment by marekweb** [comment]
If it were just disabling the alerts for the browser extension then I would suggest `hideBrowserExtensionAlerts`.

But yes, since it's also showing a message that talks about the native integration being enabled, I agree it's hard to pick a better name.

Should it be called `codeHost.nativeIntegrationEnabled` in the docs as well? rather than `codeHost.nativeIntegrationInstalled`?

**Review by marekweb — APPROVED** [review]
The new setting name is great, it communicates a lot of clarity 🎉

**Pull request merged**